src/Managers/DownloadManager.php

```diff
@@ -128,19 +137,13 @@ public function downloadRelease(array $versions)
                 array($executableName)
             );
 
-            $downloader = $this->downloadManager->getDownloaderForInstalledPackage($package);
-
-            if ($downloader === null) {
-                continue;
-            }
+            SyncHelper::downloadAndInstallPackageSync(
+                $this->composer->getLoop(),
+                $this->composer->getDownloadManager()->getDownloader('zip'),
+                $fullPath,
+                $package
+            );
 
-            /**
-             * Some downloader types have the option to mute the output,
-             * which is why there is the third call argument (not present
-             * in interface footprint).
-             */
-            $downloader->download($package, $fullPath, false);
```

Comment:
For us, this was the tricky part, that tripped me up for a while.. On Composer 1, the `getDownloaderForInstalledPackage` actually _downloads_ the package and puts it in place. In v2 it prepares the download, and if all went well, it downloads everything to process. Which makes sense, but in this case it breaks. 

That's why we replaced it with this method. See also "DownloaderInterface changes" here: https://github.com/composer/composer/blob/2.0.0/UPGRADE-2.0.md

Reply:
Yah, I think this is a much better approach. Definitely want to pull your changes into this branch.